lambdas/gro-publish-record/src/main/java/uk/gov/di/data/lep/PublishRecord.java

```diff
@@ -2,20 +2,94 @@
+    private final ObjectMapper objectMapper;
     protected static Logger logger = LogManager.getLogger();
 
+    public PublishRecord() {
+        this(new AwsService(), new Config(), HttpClient.newHttpClient(), new Mapper().objectMapper());
+    }
+
+    public PublishRecord(AwsService awsService, Config config, HttpClient httpClient, ObjectMapper objectMapper) {
+        this.awsService = awsService;
+        this.config = config;
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+    }
+
     @Override
     @Tracing
     @Logging(clearState = true)
-    public GroJsonRecord handleRequest(GroJsonRecord event, Context context) {
-        logger.info("Received xml record");
+    public Object handleRequest(GroJsonRecord event, Context context) {
+        logger.info("Received record: {}", event.RegistrationID());
+        var authorisationToken = getAuthorisationToken();
+        postRecordToLifeEvents(event, authorisationToken);
         return null;
     }
+
+    private String getAuthorisationToken() {
+        logger.info("Sending request to get authorisation token");
+        var clientId = config.getCognitoClientId();
+        var clientSecret = awsService.getCognitoClientSecret(config.getUserPoolId(), clientId);
+        var authorisationRequest = HttpRequest.newBuilder()
+            .uri(URI.create(config.getCognitoOauth2TokenUri()))
+            .header("Content-Type", "application/x-www-form-urlencoded")
+            .POST(HttpRequest.BodyPublishers.ofString(String.format(
+                "grant_type=client_credentials&client_id=%s&client_secret=%s",
+                clientId,
+                clientSecret
+            )))
+            .build();
+
+        try {
+            var response = httpClient.send(authorisationRequest, HttpResponse.BodyHandlers.ofString());
+            return objectMapper.readValue(response.body(), CognitoTokenResponse.class).accessToken();
+        } catch (IOException | InterruptedException e) {
+            logger.error("Failed to send authorisation request");
+            Thread.currentThread().interrupt();
+            throw new AuthException("Failed to send authorisation request", e);
+        }
+    }
+
+    private void postRecordToLifeEvents(GroJsonRecord event, String authorisationToken) {
+        var request = HttpRequest.newBuilder()
+            .uri(URI.create(String.format("https://%s/events/deathNotification", config.getLifeEventsPlatformDomain())))
+            .header("Authorization", authorisationToken)
+            .POST(HttpRequest.BodyPublishers.ofString(String.format("{\"sourceId\": \"%s\"}", event.RegistrationID())))
+            .build();
+
+        try {
+            logger.info("Sending GRO record request: {}", event.RegistrationID());
+            httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            throw handleGroApiCallException(e);
```

Comment:
why is this its own function here compared to above?

Reply:
Because it repeats 4 times which is a code smell